server/py/services/api/main.py

```diff
@@ -98,8 +99,33 @@ async def _move_service_to_online(self):
             services.api.initial_data.update_default_configuration_data()
             await self._start_periodic_functions()
 
+        # For the worker, fetch and sync the system metadata from the database to ensure that the config values are
+        # correctly set.
+        else:
+            await self._sync_system_metadata()
         await self._move_mounted_services_to_online()
 
+    async def _sync_system_metadata(self):
```

Comment:
I don't think this should be an `async` function if you run everything in a threadpool.